org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/SearchableEnvironment.java

```diff
@@ -115,7 +115,7 @@ public SearchableEnvironment(JavaProject project, org.eclipse.jdt.core.ICompilat
 			|| !JavaCore.IGNORE.equals(project.getOption(JavaCore.COMPILER_PB_DISCOURAGED_REFERENCE, true));
 		this.workingCopies = workingCopies;
 		this.nameLookup = project.newNameLookup(workingCopies, excludeTestCode);
-		boolean java9plus = JavaCore.callReadOnly(() -> CompilerOptions
+		boolean java9plus = release >=JavaProject.FIRST_MULTI_RELEASE || JavaCore.callReadOnly(() -> CompilerOptions
```

Comment:
This change might be correct, but I am unable to verify this, since neither the test, nor the example from https://github.com/eclipse-jdt/eclipse.jdt.core/issues/4399 needs this change. IOW, I am unable to reproduce the bug that is being fixed here. Project and editor are error free even in stock SDK 4.37.

Note: when I import the project attached to #4399 into Eclipse without this PR, the only warning I see is regarding the unknown m2e nature, as I don't have m2e installed. I did follow the steps in #4399.